iconic/UserProfile/views.py

```diff
@@ -173,17 +174,72 @@ def put(self, request):
         else:
             return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
 
- # Method for deliting city by id  
+    # Method for deliting city by id
     def delete(self, request):
         Cities.objects.get(id=request.data['id']).delete()
         return Response(status=status.HTTP_200_OK)
 
 
-#View for getting all cities
+# View for getting all cities
 class CitiesGet(APIView):
-    
-#Method for getting all cities
+
+    # Method for getting all cities
     def get(self, request):
         cities = Cities.objects.all()
         serializer = CitiesSerializer(cities, many=True)
         return Response(serializer.data, status=status.HTTP_200_OK)
+
+
+# View for getting comment! adding, getting, changing comment by authorize user
+class CommentEdit(APIView):
+    permission_classes = (permissions.IsAuthenticated,)
+
+    # Method for getting comment by id
+    def get(self, request):
+        comment = Comment.objects.get(id=request.data['id'])
+        serializer = CommentSerializer(comment)
+        if serializer.is_valid:
+            return Response(serializer.data,
+                            status=status.HTTP_200_OK)
+        else:
+            return Response(serializer.errors,
+                            status=status.HTTP_400_BAD_REQUEST)
+
+    # Method for adding comment
+    def post(self, request):
+        data = dict(request.data)
+        serializer = CommentSerializer(data=data)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data,
+                            status=status.HTTP_201_CREATED)
+        else:
+            return Response(serializer.errors,
+                            status=status.HTTP_400_BAD_REQUEST)
+
+    # Method for changing comment
+    def put(self, request):
+        comment = Comment.objects.get(id=request.data['id'])
+        data = dict(request.data)
+        serializer = CommentSerializer(comment, data=data)
```

Comment:
A comment can only be changed by the user who created it, but in this code anyone can do it